core/src/main/scala/org/apache/spark/storage/MemoryStore.scala

```diff
@@ -208,6 +208,11 @@ private[spark] class MemoryStore(blockManager: BlockManager, memoryManager: Memo
   }
 
   override def remove(blockId: BlockId): Boolean = memoryManager.synchronized {
+    val pinCount = blockManager.getPinCount(blockId)
```

Comment:
Can you turn the response from the question I asked last time about thread safety here to a comment?

I'm not quite thing I get it. A user of the block manager with code like
if (getPinCount(id) == 0) remove(id) is not generally safe right?